src/charting/EChart.js

```diff
@@ -25,6 +25,14 @@ export default class EChart extends React.Component {
         this.echart.setOption(nextProps.options);
     }
 
+    componentDidUpdate(nextProps) {
+        if (nextProps.style !== this.props.style) {
+            this.echart.resize();
+        }
+
+        this.echart.setOption(nextProps.options);
```

Comment:
These setOption calls are updating the component. We definitely need to test if we are not calling them too frequently.